s2spy/_base_calendar.py

```diff
@@ -196,8 +204,9 @@ def get_intervals(self) -> pd.DataFrame:
         if self._mapping == "data":
             self._set_year_range_from_timestamps()
 
+        skip_years = 0 if self.allow_overlap else self._get_skip_nyears()
```

Comment:
Perhaps move `skip_years = 0 if self.allow_overlap` into `self._get_skip_nyears()` to keep all the logic together?